tests/contest/contest/src/tests/poststart/mod.rs

```diff
@@ -0,0 +1,199 @@
+/// The poststart hook should execute after the container process has started.
+/// This is validated by having both the process and hook write to the same file in sequence.
+fn get_test(test_name: &'static str) -> Test {
+    Test::new(
+        test_name,
+        Box::new(move || {
+            let id = generate_uuid();
+            let id_str = id.to_string();
+            let bundle = prepare_bundle().unwrap();
+
+            let host_output_file = get_output_file_path(&bundle);
+            let host_output_file_str = host_output_file.to_str().unwrap();
+
+            let spec = get_spec(host_output_file_str);
+            set_config(&bundle, &spec).unwrap();
+
+            create_container(&id_str, &bundle, &CreateOptions::default())
+                .unwrap()
+                .wait()
+                .unwrap();
+
+            if !is_runtime_runc() && host_output_file.exists() {
+                // runc behaviour is incorrect in this case
+                // https://github.com/opencontainers/runc/issues/4347
+                let contents = fs::read_to_string(&host_output_file)
+                    .expect("failed to read output file after create");
+                if !contents.is_empty() {
+                    let _ = delete_container(&id_str, &bundle);
+                    delete_output_file(&host_output_file);
+                    let has_poststart = contents.contains("post-start called");
+                    let has_process = contents.contains("process called");
+                    return match (has_poststart, has_process) {
+                        (true, _) => TestResult::Failed(anyhow!(
+                            "The post-start hooks MUST NOT be called before the `start` operation"
+                        )),
+                        (false, true) => TestResult::Failed(anyhow!(
+                            "The user-specified program (from process) MUST NOT be run before the `start` operation"
+                        )),
+                        (false, false) => TestResult::Failed(anyhow!(
+                            "file {:?} should not exist after create",
+                            host_output_file
+                        )),
+                    };
+                }
+            }
+
+            crate::utils::start_container(&id_str, &bundle)
+                .unwrap()
+                .wait()
+                .unwrap();
+
+            let wait_result = wait_for_file_content(
+                &host_output_file,
+                "process called",
+                std::time::Duration::from_secs(5),
+                std::time::Duration::from_millis(100),
+            );
+
+            let result = if let Err(e) = wait_result {
+                TestResult::Failed(anyhow!(
+                    "The poststart hooks MUST be invoked by the runtime\n\
+                     The runtime MUST run the user-specified program, as specified by `process`\n\
+                     Error: {}",
```

Comment:
```suggestion
                    "The container process did not ran correctly : Error: {}",
```

Technically this error is when we do not get the container process o/p, so mentioning poststart hook might be confusing.